command/src/main/java/com/iluwatar/Command.java

```diff
@@ -11,6 +11,8 @@
 
 	public abstract void undo();
 
+	public abstract void redo();

```

Comment:
just thinking...
can we make undo()/redo() (as well as execute()) return boolean instead of void, so we can't shrink if no target provided, target already shrinked (small) or target is immune. 
as result command can't be queued/dequeued if execute() fails.